src/highdicom/sr/content.py

```diff
@@ -599,3 +838,163 @@ def __init__(
                 'One of the following two arguments must be provided: '
                 '"source_images" or "source_series".'
             )
+
+    @classmethod
+    def from_seg_dataset(
+            cls,
+            dataset: Dataset,
+            segment_number: int,
+            frame_numbers: Optional[Sequence[int]] = None
+        ) -> 'ReferencedSegment':
+        """Construct the content item directly from a segmentation dataset
+
+        Parameters
+        ----------
+        dataset: pydicom.dataset.Dataset
+            dataset representing a segmentation containing the referenced
+            segment
+        segment_number: int
+            number of the segment to reference within the provided dataset
+        frame_numbers: Optional[Sequence[int]], optional
+            list of frames in the segmentation dataset to reference. If
+            not provided, the reference is assumed to apply to all frames
+            of the given segment number
+
+        Returns
+        -------
+        ReferencedSegment
+            Content item representing a reference to the segment
+
+        Notes
+        -----
+             This method will attempt to deduce source image information
+             from information provided in the segmentation instance. If
+             available, it will used information specific to the segment
+             and frame numbers (if any) provided using the Derivation
+             Image Sequence information in the frames. If this information
+             is not present in the segmentation dataset, it will instead
+             use the information in the Referenced Series Sequence, which
+             applies to all segments and frames present in the segmentation
+             instance.
+
+        """
+        if dataset.SOPClassUID != SegmentationStorage:
+            raise ValueError(
+                'Input dataset should be a segmentation storage instance'
+            )
+
+        source_images = []
```

Comment:
Why not `source_images = None`?

Reply:
Because we need to append to it later in the loop beginning:

```python
for frame_info in referenced_frame_info:
```
currently on line 1006

Reply:
I see. It's potentially error prone due to the `source_images if source_images else None`, but I don't have a better suggestion either.